libs/diag/sdk_error.go

```diff
@@ -0,0 +1,61 @@
+package diag
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+)
+
+func findApiErr(e error) *apierr.APIError {
+	for {
+		cast, ok := e.(*apierr.APIError)
+		if ok {
+			return cast
+		}
+
+		inner := errors.Unwrap(e)
+		if inner == nil {
+			break
+		}
+		e = inner
+	}
+	return nil
+}
+
+func FormatAPIErrorSummary(e error) string {
+	extra := ""
+	apiErr := findApiErr(e)
+	if apiErr != nil {
+		extra = strings.TrimSpace(fmt.Sprintf("%d %s", apiErr.StatusCode, apiErr.ErrorCode))
+		extra = " (" + extra + ")"
+	}
+
+	return e.Error() + extra
+}
+
+func FormatAPIErrorDetails(e error) string {
```

Comment:
Would be good to have a unit test for this to check the output when there is or no ResponseWrapper/Request/etc.

Reply:
good idea, added here https://github.com/databricks/cli/pull/3379/commits/512323367e519bbd127ded711ccb3e41a687dbf6